src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -23,6 +24,12 @@ public static class WebScriptHostBuilderExtension
             ILoggerFactory configLoggerFactory = rootServiceProvider.GetService<ILoggerFactory>();
 
             builder.UseServiceProviderFactory(new JobHostScopedServiceProviderFactory(rootServiceProvider, rootScopeFactory))
+                .ConfigureServices(services =>
+                {
+                    // register default configuration
+                    // must happen before the script host is added below
+                    services.ConfigureOptions<HttpOptionsSetup>();
```

Comment:
Is this to allow for overriding? It would be nice if ordering wasn't a requirement and we handled the configuration being set, doing the right thing if it was (i.e., not setting the default)

Reply:
Yes it is for overriding. I think the general pattern of applying defaults early before any user config runs makes sense. How else do you propose doing this? E.g. for a boolean like HttpOptions.DynamicThrottlesEnabled we don't know if for a particular value true/false whether this came from user config. I don't think we want to impose a nullable pattern on extension options authors?

Reply:
Setup should always run after all configuration is in place. This setup should handle setting what is coming from host.json (by applying configuration or a default)

Reply:
We discussed and decided this pattern is ok